src/main/java/com/flightstats/hub/webhook/WebhookError.java

```diff
@@ -93,6 +102,58 @@ private String getChildPath(String errorRoot, String child) {
         return limitChildren(webhook);
     }
 
+    void publishToErrorChannel(DeliveryAttempt attempt) {
+        if (attempt.getWebhook().getErrorChannelUrl() == null) return;
+
+        List<String> errors = get(attempt.getWebhook().getName());
+        if (errors.size() < 1) {
+            logger.debug("no errors found for", attempt.getWebhook().getName());
+            return;
+        }
+
+        String error = errors.get(errors.size() - 1);
+        byte[] bytes = buildPayload(attempt, error);
+        Content content = Content.builder()
+                .withContentType("application/json")
+                .withContentLength((long) bytes.length)
+                .withData(bytes)
+                .build();
+
+        try {
+            channelService.insert(getChannelName(attempt.getWebhook().getErrorChannelUrl()), content);
+        } catch (Exception e) {
+            logger.warn("unable to publish errors for " + attempt.getWebhook().getName(), e);
+        }
+    }
+
+    private byte[] buildPayload(DeliveryAttempt attempt, String error) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode root = objectMapper.createObjectNode();
+        root.put("webhookUrl", buildWebhookUrl(attempt));
+        root.put("failedItemUrl", attempt.getWebhook().getChannelUrl() + "/" + attempt.getContentPath().toUrl());
+        root.put("callbackUrl", attempt.getWebhook().getCallbackUrl());
+        root.put("numberOfAttempts", attempt.getNumber() - 1);
+        root.put("lastAttemptTime", extractTimestamp(error));
+        root.put("lastAttemptError", extractMessage(error));
+        return root.toString().getBytes();
+    }
+
+    private String buildWebhookUrl(DeliveryAttempt attempt) {
+        // todo - workaround for HubHost.getLocalNamePort and HubProperties.getAppUrl not returning usable URLs for dockerized single hub
+        String host = RequestUtils.getHost(attempt.getWebhook().getChannelUrl());
+        return host + "/webhook/" + attempt.getWebhook().getName();
+    }
+
+    private String extractTimestamp(String error) {
+        return error.substring(0, error.indexOf(" "));
+    }
+
+    private String extractMessage(String error) {
```

Comment:
Looks like the errors have a specific format. Could probably use a comment or a reference to a file with comments on what that format is.

Reply:
Good call. Since ```WebhookError.java``` is responsible for creating/parsing those strings I'll add a comment to the file indicating the format.